quinn-proto/src/connection/assembler.rs

```diff
@@ -193,6 +281,55 @@ impl Chunk {
 struct Buffer {
     offset: u64,
     bytes: Bytes,
+    size: usize,
```

Comment:
Is this ever different from `bytes.len()`? Can't exactly find the case. Could we maybe have a more descriptive name, which hints under which conditions that might be different?

Reply:
There is currently only one case in which it will differ: inside `read`, if `max_length < bytes.len()`. We didn't want to update `size` there because it could lead to defragmenting an already defragmented buffer again, but this is no longer the case. 

I am now thinking about replacing the `size: usize` and `state: BufferState` simply with `allocation_size: usize` and `fragmented: bool`. I think this would make things a lot clearer.

Reply:
The `size` field does no longer exist.